core/src/main/java/org/apache/calcite/sql/dialect/SybaseSqlDialect.java

```diff
@@ -36,11 +36,11 @@ public SybaseSqlDialect(Context context) {
 
   @Override public void unparseOffsetFetch(SqlWriter writer, SqlNode offset,
       SqlNode fetch) {
-    // No-op; see unparseTopN.
+    // No-op; see unparseFetchAsTopN.
```

Comment:
I might have missed it but where is FETCH NEXT n ROWS -> SELECT TOP(n) tested?

Reply:
I believe this is tested in the RelToSqlConverterTest file (for example in testSelectQueryWithFetchClause unit test). Converting FETCH statements into SELECT TOP N statements is not actually the functionality we are trying to add here, so the changes here are just to disentangle the TOP N unparsing logic from the FETCH unparsing logic (which sometimes ends up being unparsed as a TOP N statement as per pre-existing code).

Reply:
That makes sense